src/sentry/static/sentry/app/views/releaseDetails.jsx

```diff
@@ -18,6 +21,54 @@ import TimeSince from 'app/components/timeSince';
 import Version from 'app/components/version';
 import withEnvironmentInQueryString from 'app/utils/withEnvironmentInQueryString';
 
+class ReleaseDetailsActions extends React.Component {
+  static propTypes = {
+    api: PropTypes.object.isRequired,
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    release: PropTypes.object.isRequired,
+  };
+
+  handleDelete = () => {
+    let {orgId, projectId, release, api} = this.props;
+    let version = encodeURIComponent(release.version);
+    let path = `/organizations/${orgId}/releases/${version}/`;
+    let redirectPath = `/${orgId}/${projectId}/releases/`;
+    let indicator = IndicatorStore.add(t('Deleting Release...'));
+
+    api.request(path, {
+      method: 'DELETE',
+      success: () => {
+        IndicatorStore.remove(indicator);
+        browserHistory.push(redirectPath);
+      },
+      error: err => {
+        IndicatorStore.remove(indicator);
+        IndicatorStore.addError(t(err.responseJSON.detail));
+      },
+    });
+  };
+
+  render() {
+    return (
+      <div className="release-actions m-b-1">
+        <div className="btn-group">
+          <LinkWithConfirmation
+            className="release-remove btn btn-default btn-sm"
+            title={t('Delete')}
+            message={t(
+              'Deleting this release is permanent. Are you sure you wish to continue?'
+            )}
+            onConfirm={this.handleDelete}
+          >
+            <span className="icon-trash" style={{marginRight: 5}} />Delete
```

Comment:
Hmm, now looking at this, we might actually want [Button](https://storybook.getsentry.net/branches/master/index.html?knob-Extra%20Small%20Busy=true&knob-Small%20Busy=true&knob-Normal%20Busy=true&knob-Large%20Busy=true&selectedKind=Buttons&selectedStory=overview&full=0&down=1&left=1&panelRight=0&downPanel=storybooks%2Fstorybook-addon-knobs) here since it'll use our updated [trash icon](https://storybook.getsentry.net/branches/master/index.html?selectedKind=Icons&selectedStory=SVG&full=0&down=1&left=1&panelRight=0&downPanel=storybooks%2Fstorybook-addon-knobs) and we can just remove the class names from `LinkWithConfirmation`